testing/functional/tests/cypress/integration/institute/03-institution-profile.spec.ts

```diff
@@ -0,0 +1,141 @@
+    cy.visit(url);
+    cy.intercept("GET", "**/bceid-account").as("bceidAccount");
+    loginInstituteObject.loginWithBCEID().should("be.visible").click();
+    loginInstituteObject.loginInWithBCEIDtext().should("be.visible");
+    loginInstituteObject
+      .bceidInputText()
+      .type(username)
+      .should("have.value", username);
+    loginInstituteObject
+      .passwordInputText()
+      .type(password)
+      .should("have.value", password);
+    loginInstituteObject.continueButton().click();
+    cy.wait("@bceidAccount");
+  });
+
+  it("Verify that user redirect to institute profile page", () => {
+    dashboardInstituteObject.dashboardButton().click();
+    dashboardInstituteObject.manageInstitutionButton().click();
+    dashboardInstituteObject.locationVerifyText().should("be.visible");
+    institutionObject.institutionDetailsButton().click();
+    institutionObject.institutionInformationText().should("be.visible");
+  });
+
+  it("Check that legal operating name text field should be disable or not", () => {
+    dashboardInstituteObject.dashboardButton().click();
+    dashboardInstituteObject.manageInstitutionButton().click();
+    dashboardInstituteObject.locationVerifyText().should("be.visible");
+    institutionObject.institutionDetailsButton().click();
+    institutionObject.legalOperatingNameInputText().should("be.disabled");
+  });
+
+  it("Clicking on the save button without filling out the required fields", () => {
+    dashboardInstituteObject.dashboardButton().click();
+    dashboardInstituteObject.manageInstitutionButton().click();
+    dashboardInstituteObject.locationVerifyText().should("be.visible");
+    institutionObject.institutionDetailsButton().click();
+    institutionObject.primaryEmailInputText().clear();
+    cy.wait(2000);
+    institutionObject.submitButton().click();
+  });
+
+  it(
+    "Check that when user enter data only in non mandatory field and click on next section.",
+    { retries: 4 },
+    () => {
+      dashboardInstituteObject.dashboardButton().click();
+      dashboardInstituteObject.manageInstitutionButton().click();
+      dashboardInstituteObject.locationVerifyText().should("be.visible");
+      institutionObject.institutionDetailsButton().click();
+      cy.reload();
+      institutionObject.primaryEmailInputText().clear();
+      institutionObject.primaryEmailInputText().clear();
+      cy.wait(2000);
+      institutionObject.submitButton().click();
+      institutionObject.unexpectedErrorMessage().should("be.visible");
+    }
+  );
+
+  it("Check when user enter data only in mandatory field then able to save the details or not.", () => {
+    cy.fixture("institutionProfileData").then((data) => {
+      dashboardInstituteObject.dashboardButton().click();
+      dashboardInstituteObject.manageInstitutionButton().click();
+      dashboardInstituteObject.locationVerifyText().should("be.visible");
+      institutionObject.institutionDetailsButton().click();
+      cy.wait(1000);
```

Comment:
Is there a reason to use the `cy.wait(1000)` with a specific timeout here?

Reply:
When a page is redirected to a form page, all input fields are automatically added, so the selector takes milliseconds to recognize the selector in the form page (Input Text, Dropdown, Radio Button etc.). If I remove this wait. When it redirects to the form page, two of five times the selector is not found, and an error message appears, so I have to wait a few seconds. Pages sometimes load within seconds but sometimes they don't because of a delayed response from the server.

Reply:
My comment was not about removing the `wait`, it was about why wait at a specific time. As mentioned before it is mentioned as a bad practice, mostly because if the website takes more than 1000 t respond it will end up in a false positive.
To summarize, the wait is okay, wait at a specific time does not seem desirable.

Reply:
Would it be okay this wait in common for everywhere (2000ms) ?
<img width="631" alt="image" src="https://user-images.githubusercontent.com/98920571/163014778-c378013f-6dbd-4766-ba59-bfd3bef80577.png">
<img width="278" alt="image" src="https://user-images.githubusercontent.com/98920571/163014897-457639ee-14a7-4f94-a176-b6f07046201b.png">

Reply:
When I am mentioning to avoid using the `cy.wait` it is because so far everything leads me to think that it should be used when there is not an alternative or looking for an alternative would take a lot of time and it would not be worth. As stated below in the comment of the `cy.wait`.
![image](https://user-images.githubusercontent.com/61259237/163054287-0d2cbb33-dad7-4ae3-a13d-f34d385c7f2c.png)

For instance the `.contains`, as per the documentation (https://docs.cypress.io/api/commands/contains#Syntax) will try to find the component already considering the `defaultCommandTimeout` so there is already a wait mechanism in place.